src/main/java/levi9/UserService/controller/UserController.java

```diff
@@ -24,12 +25,19 @@ public UserController(UserService userService) {
         return ResponseEntity.ok(userService.register(newUser));
     }
 
-
-    @GetMapping(value = "/{username}")
+    //get user by its username
+    @GetMapping(value = "/getUser/{username}")
```

Comment:
instead of getUser use just user (/user/{username}. When it is GET it is clear that you are retrieving resource

Reply:
solved